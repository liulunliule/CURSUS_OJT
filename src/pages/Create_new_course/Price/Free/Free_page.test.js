import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Free_page from '.';
import { Switch } from 'antd';

describe('Free_page Component', () => {
  // Snapshot Test
  test('matches snapshot', () => {
    const tree = renderer
      .create(<Free_page />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Unit Tests
  test('renders switches and text correctly', () => {
    // Render the Free_page component
    render(<Free_page />);

    // Check if both switches are present
    const switchElements = screen.getAllByRole('switch');
    expect(switchElements).toHaveLength(2); // There should be 2 switches

    // Check if the text content is rendered correctly
    expect(screen.getByText('Require Log in')).toBeInTheDocument();
    expect(screen.getByText('Require Enroll')).toBeInTheDocument();
    expect(screen.getByText(/If the course is free, if student require to enroll your course,/)).toBeInTheDocument();
  });
});
