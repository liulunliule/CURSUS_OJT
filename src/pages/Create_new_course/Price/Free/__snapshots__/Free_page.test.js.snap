// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Free_page Component matches snapshot 1`] = `
<div
  className="free_page"
>
  <div
    className="free_page_switch-btn"
  >
    <button
      aria-checked={false}
      className="ant-switch css-dev-only-do-not-override-1uq9j6g"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="switch"
      style={
        Object {
          "marginRight": "10px",
        }
      }
      type="button"
    >
      <div
        className="ant-switch-handle"
      />
      <span
        className="ant-switch-inner"
      >
        <span
          className="ant-switch-inner-checked"
        />
        <span
          className="ant-switch-inner-unchecked"
        />
      </span>
    </button>
    Require Log in
  </div>
  <div
    className="free_page_switch-btn"
  >
    <button
      aria-checked={false}
      className="ant-switch css-dev-only-do-not-override-1uq9j6g"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="switch"
      style={
        Object {
          "marginRight": "10px",
        }
      }
      type="button"
    >
      <div
        className="ant-switch-handle"
      />
      <span
        className="ant-switch-inner"
      >
        <span
          className="ant-switch-inner-checked"
        />
        <span
          className="ant-switch-inner-unchecked"
        />
      </span>
    </button>
    Require Enroll
  </div>
  <p
    className="free_page-content"
  >
    If the course is free, if student require to enroll your course, if not required enroll, if students required sign in to your website to take this course.
  </p>
</div>
`;
